Sources/NIOResumableUpload/HTTPResumableUploadHandler.swift

```diff
@@ -0,0 +1,141 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIOCore
+import NIOHTTPTypes
+
+/// A channel handler that translates resumable uploads into regular uploads, and passes through
+/// other HTTP traffic.
+public final class HTTPResumableUploadHandler: ChannelDuplexHandler {
+    public typealias InboundIn = HTTPRequestPart
+    public typealias InboundOut = Never
+
+    public typealias OutboundIn = Never
+    public typealias OutboundOut = HTTPResponsePart
+
+    var upload: HTTPResumableUpload
+
+    private var context: ChannelHandlerContext!
+    private var eventLoop: EventLoop!
+
+    /// Create an `HTTPResumableUploadHandler` within a given `HTTPResumableUploadContext`.
+    /// - Parameters:
+    ///   - context: The context for this upload handler.
+    ///              Use the same context across upload handlers, as uploads can't resume across different contexts.
+    ///   - channelConfigurator: A closure for configuring the child HTTP server channel.
+    public init(
+        context: HTTPResumableUploadContext,
+        channelConfigurator: @escaping (Channel) -> Void
```

Comment:
We normally call this the `channelInitializer`. Furthermore we normally let this return an `EventLoopFuture` to give people the chance to initialize their channel asynchronously.

Reply:
I checked with @Lukasa before about this. Making it asynchronous is possible but it would definitely complicate the code by adding a pause after completing the header read. This is guaranteed to be invoked no more than once, so they can do the expensive work upfront and capture the result into this closure.

Reply:
I think at least we should rename it to `channelInitializer`. I am still a bit worried about diverting from the norm here but if @Lukasa is happy with it we can go ahead.

Reply:
Yeah, we could change that. The terminology `configurator` came from nio-http2: https://github.com/apple/swift-nio-http2/blob/a6c66bcbe7dfba6f23bff8599376b5fad858a818/Sources/NIOHTTP2/HTTP2PipelineHelpers.swift#L336